src/api/index.js

```diff
@@ -0,0 +1,272 @@
+import config from '../config'
+import {mergeRecursive} from "../helpers/mergeObjects";
+import redirect from "react-router/es/Redirect";
+
+
+class ApiClass {
+    constructor() {
+        this.url = `${config.apiUrl}`;
+    }
+
+    makeRequest = ({url, data, token}) => {
+        const _data = {
+            headers: {
+                'Accept': 'application/json',
+                'Content-Type': 'application/json',
+                ...!!token && {Authorization: `Bearer ${token}`},
+
+            },
+            credentials: 'include',
+
+        };
+
+        const mergedData = mergeRecursive(data, _data);
+
+        return fetch(url, mergedData)
```

Comment:
```suggestion
        return fetch(`${this.url}/url, mergedData)
```
with this solution you will reduce your code in all of methods (in this case you won't need to add this.url in all methods.
what do you think?